setup.pl

```diff
@@ -674,11 +697,26 @@
 		print RELOADD "$systemctlcmd reload webmin\n";
 		close(RELOADD);
 
+		# Pre install
+		open(PREINSTT, ">$config_directory/.pre-install");
+		print PREINSTT "#!/bin/sh\n";
+		print PREINSTT "$systemctlcmd kill --signal=SIGSTOP --kill-who=main webmin\n";
```

Comment:
Please remove this sending of STOP / CONT signals from this PR, unless it is 100% necessary to fix the restart issue. I understand why we want it in theory, but it should be a separate PR.

Reply:
This is not strictly necessary, and most likely would work most of the times. However, removing it may be risky, as 
 the main process will be accessible (as explained earlier). Also, if we removed it the new logic won't replicate old behaviour, where Webmin process was stopped before running upgrade.

I don't disagree that it could be a separate PR, however I cannot agree, that this change should go to the release version after upcoming (like 1.997).

Reply:
Let's make it a separate PR. Also, if another user does access Webmin during the upgrade process, will this actually cause the upgrade to fail?

Reply:
> Let's make it a separate PR. 

.. ohh, okay .. let's ..

> Also, if another user does access Webmin during the upgrade process, will this actually cause the upgrade to fail?

If the process is on-pause or stopped another user won't even be able to access it. But even if it's not suspended, I don't think that just accessing Webmin would do any harm.

Reply:
There you go - https://github.com/webmin/webmin/pull/1677/commits/6f5bb3947f3eee55c338d9902161c928a4cd8ee4.